packages/next/build/webpack/plugins/webpack-conformance-plugin/index.ts

```diff
@@ -0,0 +1,153 @@
+import { Compiler, compilation } from 'webpack'
+import {
+  IConformanceTestResult,
+  IWebpackConformanceTest,
+  IConformanceAnomaly,
+  IGetAstNodeResult,
+  NodeInspector,
+} from './TestInterface'
+import { NodePath } from 'ast-types/lib/node-path'
+import { visit } from 'recast'
+
+export { MinificationConformanceCheck } from './checks/minification-conformance-check'
+// export { ReactSyncScriptsConformanceTest } from './tests/react-sync-scripts-conformance';
+
+export interface IWebpackConformancePluginOptions {
+  tests: IWebpackConformanceTest[]
+}
+
+interface VisitorMap {
+  [key: string]: (path: NodePath) => void
+}
+
+export default class WebpackConformancePlugin {
+  private tests: IWebpackConformanceTest[]
+  private errors: Array<IConformanceAnomaly>
+  private warnings: Array<IConformanceAnomaly>
+  private compiler?: Compiler
+
+  constructor(options: IWebpackConformancePluginOptions) {
+    this.tests = []
+    if (options.tests) {
+      this.tests.push(...options.tests)
+    }
+    this.errors = []
+    this.warnings = []
+  }
+
+  private gatherResults(results: Array<IConformanceTestResult>): void {
+    results.forEach(result => {
+      if (result.result === 'FAILED') {
+        result.errors && this.errors.push(...result.errors)
+        result.warnings && this.warnings.push(...result.warnings)
+      }
+    })
+  }
+
+  private buildStartedHandler = (
+    compilation: compilation.Compilation,
+    callback: () => void
+  ) => {
+    const buildStartedResults: IConformanceTestResult[] = this.tests.map(
+      test => {
+        if (test.buildStared && this.compiler) {
+          return test.buildStared(this.compiler.options)
+        }
+        return {
+          result: 'SUCCESS',
+        } as IConformanceTestResult
+      }
+    )
+
+    Promise.all(buildStartedResults).then(
```

Comment:
Is this intentionally not awaited before `callback()` is invoked?

Reply:
this was a premature optimization i was trying. removed it for now